test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -223,6 +226,161 @@ func (data *testData) pingBetweenInterfaces(t *testing.T) error {
+			return nil, fmt.Errorf("failed to parse NetworkAttachmentDefinition config JSON for interface %s: %w", iface, err)
+		}
+
+		ipPoolsMap[iface] = netAttachDefConfig.IPAM.IPPools
+
+	}
+	return ipPoolsMap, nil
+}
+
+func (data *testData) checkIPReleased(ipPools map[string][]string, ifacesIPs map[string]net.IP, ifaces []string) error {
+	crdClient := data.e2eTestData.CRDClient
+
+	return wait.PollUntilContextTimeout(context.Background(), time.Second, 10*time.Second, true, func(ctx context.Context) (bool, error) {
+		for _, iface := range ifaces {
+			ipPoolName, exists := ipPools[iface]
+			if !exists {
+				return false, fmt.Errorf("no IPPool found for interface %s", iface)
+			}
+
+			podIP, exists := ifacesIPs[iface]
+			if !exists {
+				return false, fmt.Errorf("no IP found for interface %s", iface)
+			}
+
+			ipPool, err := crdClient.CrdV1beta1().IPPools().Get(ctx, ipPoolName[0], metav1.GetOptions{})
+			if err != nil {
+				return false, fmt.Errorf("failed to get IPPool %s: %w", ipPoolName[0], err)
+			}
+
+			for _, ipAddress := range ipPool.Status.IPAddresses {
+				if podIP.String() == ipAddress.IPAddress {
+					return false, nil
+				}
+			}
+			logs.Infof("Released PodIP: %v, Interface: %s", podIP, iface)
+		}
+		return true, nil
+	})
+}
+
+func checkOVSPorts(before, after []string) bool {
+	afterSet := make(map[string]struct{})
+	for _, port := range after {
+		afterSet[port] = struct{}{}
+	}
+
+	for _, port := range before {
+		if _, exists := afterSet[port]; !exists {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (data *testData) getOVSPortsOnSecondaryBridge(t *testing.T, nodeName string) ([]string, error) {
+	cmd := []string{"ovs-vsctl", "list-ports", "br-secondary"}
+
+	stdout, stderr, err := data.e2eTestData.RunCommandFromAntreaPodOnNode(nodeName, cmd)
+	if err != nil {
+		return nil, fmt.Errorf("failed to run ovs-vsctl on node %s: %v\nstderr: %s", nodeName, err, stderr)
+	}
+
+	ovsPorts := strings.Fields(stdout)
+	return ovsPorts, nil
+}
+
+// reconcilationAfterAgentRestart verifies OVS Ports cleanup and IP release.
+func (data *testData) reconcilationAfterAgentRestart(t *testing.T) error {
+	nodes := []string{data.pods[0].nodeName, data.pods[2].nodeName}
+	for _, node := range nodes {
+		ovsPortsBeforeAgentRestart, err := data.getOVSPortsOnSecondaryBridge(t, node)
+		if err != nil {
+			return fmt.Errorf("failed to get Secondary bridge OVS Ports before agent restart: %w", err)
+		}
+
+		// Restarting the Antrea agent
+		if err := data.e2eTestData.RestartAntreaAgentPods(30 * time.Second); err != nil {
+			t.Fatalf("Failed to restart Antrea agent Pods: %v", err)
+		}
+
+		ovsPortsAfterAgentRestart, err := data.getOVSPortsOnSecondaryBridge(t, node)
+		if err != nil {
+			return fmt.Errorf("failed to get Secondary bridge OVS Ports after agent restart: %w", err)
+		}
+
+		if !checkOVSPorts(ovsPortsBeforeAgentRestart, ovsPortsAfterAgentRestart) {
```

Comment:
Still good to keep the earlier code of checking interfaces and IPs. Why remove that code?

Reply:
I thought we need to just check ovs ports , again I added that, thanks!